bluemira/base/reactor.py

```diff
@@ -129,12 +153,34 @@ def show_cad(self, *dims, **kwargs):
                     f"Must be one of {str(_PLOT_DIMS)}"
                 )
 
-        comp = self.component()
+        comp = self.component(with_components)
 
-        # Filtering mutates the underlying components,
-        # which exist in memory regardless of calling self.component() on the reactor.
-        # Thus a copy must be made
+        # A copy of the component tree must be made
+        # as filtering would mutate the underlying component tree,
+        # that exists statically in memory.
+        # self.component (above) only creates a new root node for this reactor,
+        # not a new component tree.
         comp_copy = comp.copy()
         comp_copy.filter_components(dims_to_show)
 
+        if "xyz" in dims_to_show:
+            xyzs = comp_copy.get_component(
+                "xyz",
+                first=False,
+            )
+            xyzs = [xyzs] if isinstance(xyzs, Component) else xyzs
+
+            comp_names = (
+                "all components"
+                if not with_components
+                else ", ".join([cm.component().name for cm in with_components])
+            )
+            print(f"Building xyz CAD with {with_n_sectors} sectors for {comp_names}")
```

Comment:
We have explicitly avoided print everywhere could you use either `bluemira_print` or `bluemira_print_clean` so that we can have a silent terminal in future if required by setting a log level. 

Probably need a think with the progress bar how we could achieve the same thing either way for now thats fine for me